sources/Inputs/include/AssemblingXmlDocument.h

```diff
@@ -0,0 +1,305 @@
+//
+// Copyright (c) 2021, RTE (http://www.rte-france.com)
+// See AUTHORS.txt
+// All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, you can obtain one at http://mozilla.org/MPL/2.0/.
+// SPDX-License-Identifier: MPL-2.0
+//
+
+/**
+ * @file AssemblingXmlDocument.h
+ * @brief Assembly xml document handler header
+ */
+
+#pragma once
+
+#include <boost/optional.hpp>
+#include <string>
+#include <vector>
+#include <xml/sax/parser/Attributes.h>
+#include <xml/sax/parser/ComposableDocumentHandler.h>
+#include <xml/sax/parser/ComposableElementHandler.h>
+
+namespace dfl {
+namespace inputs {
+
+/**
+ * @brief Assembly xml document handler
+ *
+ * XML document handler for the assembling file for dynamic models
+ */
+class AssemblingXmlDocument : public xml::sax::parser::ComposableDocumentHandler {
+ public:
+  /**
+   * @brief Connection XML element
+   */
+  struct Connection {
+    std::string var1;  ///< first variable to connect
+    std::string var2;  ///< second variable to connect
+  };
+
+  /**
+   * @brief Bus XML element
+   */
+  struct Bus {
+    std::string voltageLevel;  ///< voltage level of the bus
+  };
+
+  /**
+   * @brief Shunt XML element
+   */
+  struct Shunt {
+    std::string voltageLevel;  ///< id of the voltage level containing the shunts
+  };
+
+  /**
+   * @brief Transfo XML element
+   */
+  struct Tfo {
+    std::string name;  ///< name of the transfo
+  };
+
+  /**
+   * @brief Line XML element
+   */
+  struct Line {
+    std::string name;  ///< name of the line
+  };
+
+  /**
+   * @brief Macro connect XML element defining a macro connection of a dynamic model
+   */
+  struct MacroConnect {
+    std::string macroConnection;  ///< macro connection id
+    std::string id;               ///< association id to use
+  };
+
+  /**
+   * @brief Macro connection XML element defining a macro connector
+   */
+
+  struct MacroConnection {
+    std::string id;                       ///< macro connection id
+    std::vector<Connection> connections;  ///< list of connections of the macro connections
+  };
+
+  /**
+   * @brief Single association XML element
+   */
+  struct SingleAssociation {
+    std::string id;              ///< association id
+    boost::optional<Bus> bus;    ///< bus of the association
+    boost::optional<Tfo> tfo;    ///< transfo of the association
```

Comment:
in comment, transformer instead of transfo

Reply:
OK